src/backends/torch/native/templates/tt/tdecoder.cc

```diff
@@ -0,0 +1,95 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2019-2020 Jolibrain
+ * Author:  Guillaume Infantes <guillaume.infantes@jolibrain.com>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "mllibstrategy.h"
+
+#include "tdecoder.h"
+
+namespace dd
+{
+  void TDecoderImpl::init()
+  {
+    if (_simple)
+      {
+        _lins.push_back(register_module(
+            "decoder_0", torch::nn::Linear(_embed_dim * _input_len,
+                                           _output_dim * _output_len)));
+        for (int i = 1; i < _nlayers; ++i)
+          {
+            std::string lname = "decoder_" + std::to_string(i);
+            _lins.push_back(register_module(
+                lname, torch::nn::Linear(_output_dim * _output_len,
+                                         _output_dim * _output_len)));
+          }
+        if (_dropout_ratio != 0.0)
+          _dropout = register_module("decoder_dropout",
+                                     torch::nn::Dropout(_dropout_ratio));
+      }
+    else
+      {
+        torch::nn::TransformerDecoderLayer decoder_layer
+            = torch::nn::TransformerDecoderLayer(
+                torch::nn::TransformerDecoderLayerOptions(_output_dim, _nheads)
+                    .dropout(_dropout_ratio)
+                    .dim_feedforward(_hidden_dim)
+                    .activation(torch::kReLU));
```

Comment:
Transformer litterature seems to be using GeLU, any interest here instead of ReLU ?

Reply:
fixed